official/utils/logging/logger.py

```diff
@@ -72,3 +84,106 @@ def log_metric(self, name, value, unit=None, global_step=None, extras=None):
       except (TypeError, ValueError) as e:
         tf.logging.warning("Failed to dump metric to log file: "
                            "name %s, value %s, error %s", name, value, e)
+
+  def log_run_info(self, model_name):
+    """Collect most of the TF runtime information for the local env.
+
+    The schema of the run info follows official/benchmark/datastore/schema.
+    """
+    run_info = {"model_name": model_name}
+    _collect_tensorflow_info(run_info)
+    _collect_environment_variable(run_info)
+    _collect_cpu_info(run_info)
+    _collect_gpu_info(run_info)
+    _collect_memory_info(run_info)
+
+    with tf.gfile.GFile(os.path.join(
+        self._logging_dir, _BENCHMARK_RUN_LOG_FILE_NAME), "w") as f:
```

Comment:
Does this assume this is the first thing that gets run? Why not append mode here?

Reply:
The metric info is logged to a different file, which gets appended, whereas the run info is a big blob of JSON, which is stored in a single file. Each run should have one run info, and append run info kind of confuse the uploading part in future.